cranelift/codegen/src/isa/x64/inst/emit.rs

```diff
@@ -1048,46 +1050,40 @@ pub(crate) fn emit(inst: &Inst, sink: &mut MachBuffer<Inst>) {
                 RexFlags::clear_w(),
             );
         }
-
-        Inst::XMM_RM_R {
+        Inst::XMM_MOV_RM_R {
+            op,
+            src: srcE,
+            dst: reg_g,
+        }
```

Comment:
Even if it is a bit redundant, how about deduplicating the bodies for these two variants? Then we could more effectively separate these by opcodes, and catch invalid (Inst kind, opcode) combinations.